test/unit/iteration_test.rb

```diff
@@ -219,6 +236,12 @@ def test_jobs_using_complex_but_serializable_cursor_will_not_raise
     work_one_job
   end
 
+  def test_jobs_using_on_complete_have_accurate_total_time
+    freeze_time do
+      JobThatCompletesAfter3Seconds.perform_now(->(job) { assert_equal(3, job.total_time) })
+    end
+  end
```

Comment:
I would suggest simplifying the test:

```ruby
def test_jobs_using_on_complete_have_accurate_total_time
    freeze_time do
      job = JobThatCompletesAfter3Seconds.new
      job.perform_now
      assert_equal(3, job.total_time)
    end
  end
```

That way you do not need to pass the assertion around, which is a little weird 😄